### DISCUSSION TITLE
Is the Docker image using the correct chromedriver?

### DISCUSSION THREAD

**Discussion by wendigo1234** [jdholtz/auto-southwest-check-in#250] [General]
I'm not super knowledgeable about Selenium, but it looks like this script is using the regular chromedriver instead of the undetectable chromedriver?

```
RUN cp /usr/bin/chromedriver /usr/local/lib/python3.12/site-packages/seleniumbase/drivers/uc_driver
```

Isn't uc_driver something Selenium normally downloads on its own? Or am I missing something? Is it inadvisable to run this script from a Raspberry Pi even with the Docker image due to the potential detection problem?

**Comment by MarouaneBenbetka** [comment]
No it's not downloaded by default , you need to download it .

Running inside Docker on a Raspberry Pi should not impact the detectability of your automated browsing as long as you're using the correct version of the driver and your Selenium scripts are properly configured.

i usually use chrome_undeteced_driver with a chrome profile to not get detected like this :
```
import undetected_chromedriver as uc
PATH_CHROME_PROFILE = r"__PRIVATE_INFO_HHH_"
options = uc.ChromeOptions()
options.add_argument(
    r"--user-data-dir="+PATH_CHROME_PROFILE)
```

if this answers your question mark it as answered ,Good luck.

**Comment by jdholtz** [comment]
> I'm not super knowledgeable about Selenium, but it looks like this script is using the regular chromedriver instead of the undetectable chromedriver?

SeleniumBase downloads the chromedriver automatically, but that doesn't work on Alpine Linux which is what the Docker image uses (it especially doesn't work on ARMv64 architecture). Therefore, the Docker image just downloads the APK package and manually copies it to the correct place. SeleniumBase will patch the driver to make it more undetectable if it isn't so it is not a worry that it is copied manually.

>  Is it inadvisable to run this script from a Raspberry Pi even with the Docker image due to the potential detection problem?

As @MarouaneBenbetka said above, running on a Raspberry Pi should not make you any more detectable than any other machine.

**Reply by jdholtz** [reply]
Hey @wendigo1234, does my reply above (or MarouaneBenbetka's) answer your question?
